x/ratelimit/keeper/rate_limit.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v4/x/ratelimit/types"
+)
+
+// Get the rate limit byte key built from the denom and channelId
+func GetRateLimitItemKey(denom string, channelId string) []byte {
+	return append(types.KeyPrefix(denom), types.KeyPrefix(channelId)...)
+}
+
+// The total value on a given path (aka, the denominator in the percentage calculation)
+// is the total supply of the given denom
+func (k Keeper) GetChannelValue(ctx sdk.Context, denom string) sdk.Int {
```

Comment:
I recall Osmosis summing up across multiple channels - why were they doing that and why don't we have to account for that here?

Reply:
Osmosis did not sum across multiple channels - they just used the total supply. They mentioned that they _wanted_ to  sum across multiple channels but the functionality wasn't present in the SDK.

Their rationale was that they wanted to keep quota thresholds for the same denom relatively similar. For instance, there aren't a lot of people sending their `uosmo` to the hub, but there might be a lot that are sending it to Stride. So the quota thresholds on those two channels would have to be quite different (since the denominators are of different magnitudes), even though the denom is the same.

And they also felt that using the total escrow balance across each channel was more accurate than using the total supply since most native tokens were kept on Osmosis (as they had little utility elsewhere). 
Meaning: `Total Supply >>> Total Escrow `

In our case, most of our native tokens (stTokens) will be moved off Stride right away (since they have little utility on Stride), so the total supply is likely much closer to the total escrow balance and thus is a more accurate metric for us.
Meaning: `Total Supply ~= Total Escrow`